R/test-files.r

```diff
@@ -77,26 +123,42 @@ source_dir <- function(path, pattern = "\\.[rR]$", env = test_env(),
 #' @param path path to file
 #' @param reporter reporter to use
 #' @param env environment in which to execute the tests
-#' @return a data frame of the summary of test results
+#' @return the results as a "testthat_results" (list)
 #' @export
 test_file <- function(path, reporter = "summary", env = test_env()) {
-  current_reporter <- find_reporter(reporter)
+  reporter <- if (is.null(reporter)) NULL else find_reporter(reporter)
```

Comment:
because it's your function, and I did not want to potentially break existing code.